### PR TITLE
Next.js Demo: Readme.md revamp

### PR DESCRIPTION
**Pull request by jmderby**
## Description

View the readme here: https://github.com/Crossmint/crossmint-sdk/blob/jonathan/nextjs-readme/apps/wallets/smart-wallet/next/README.md

Update the README to provide a more comprehensive and user-friendly guide for the Smart Wallets Demo (Next.js Starter Kit).

## Test plan

Readme is better than the original

### CODE REVIEW

File: apps/wallets/smart-wallet/next/README.md [hunk 1]
```diff
@@ -1,40 +1,155 @@
...
+        <CrossmintAuthProvider
+            apiKey="YOUR_CROSSMINT_API_KEY"
+            embeddedWallets={{
+                createOnLogin: "all-users",
+                defaultChain: "polygon-amoy",
+                type: "evm-smart-wallet",
+            }}
+        >
+            <Component {...pageProps} />
+        </CrossmintAuthProvider>
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/next/README.md:134]: This currently wouldn't work right?

Comment (devlyn37) [reply apps/wallets/smart-wallet/next/README.md:134]: Since, for the time being, we require the environment prop too

**Review by devlyn37 — APPROVED** [review]
Approved w/ important comment!

File: apps/wallets/smart-wallet/next/README.md (hunk 1 above)
Comment (jmderby) [reply apps/wallets/smart-wallet/next/README.md]: I've added the env prop in the example and made a reminder in the `environment` prop removal ticket to also remove it from the example.

**Pull request merged**